src/GafferOSL/OSLObject.cpp

```diff
@@ -57,14 +61,31 @@ size_t OSLObject::g_firstPlugIndex;
 namespace
 {
 
-CompoundDataPtr prepareShadingPoints( const Primitive *primitive )
+
+CompoundDataPtr prepareShadingPoints( const Primitive *primitive, const std::string &requiredPrimvars )
 {
+	std::vector<std::string> primvarNames;
+	boost::algorithm::split( primvarNames, requiredPrimvars, boost::algorithm::is_any_of(" ") );
+
 	CompoundDataPtr shadingPoints = new CompoundData;
 	for( PrimitiveVariableMap::const_iterator it = primitive->variables.begin(), eIt = primitive->variables.end(); it != eIt; ++it )
 	{
-		// cast is ok - we're only using it to be able to reference the data from the shadingPoints,
-		// but nothing will modify the data itself.
-		shadingPoints->writable()[it->first] = boost::const_pointer_cast<Data>( it->second.data );
+		bool primvarUsed = std::find( primvarNames.begin(), primvarNames.end(), it->first ) != primvarNames.end();
```

Comment:
Any quadratic complexity concerns here? Do we expect the number of primvars to always be low enough to get away with linear search?